src/Fantomas/Program.fs

```diff
@@ -325,27 +382,54 @@ let main argv =
 
             processFile force i o)
 
-        Seq.length files
-
-    let filesAndFolders force (files: string list) (folders: string list) : int =
-        let singleFilesProcessed =
+    let filesAndFolders force (files: string list) (folders: string list) =
+        let fileTasks =
             files
-            |> List.sumBy (fun file ->
+            |> List.map (fun file ->
                 if (IgnoreFile.isIgnoredFile (IgnoreFile.current.Force()) file) then
                     logGrEqDetailed verbosity $"'%s{file}' was ignored"
-                    0
+                    async.Return(ProcessResult.Ignored(file))
                 else
-                    processFile force file file
-                    1)
+                    processFile force file file)
 
-        let filesInFoldersProcessed =
-            folders |> List.sumBy (fun folder -> processFolder force folder folder)
+        let folderTasks =
+            folders |> List.collect (fun folder -> processFolder force folder folder)
 
-        singleFilesProcessed + filesInFoldersProcessed
+        (fileTasks @ folderTasks)
 
     let check = results.Contains <@ Arguments.Check @>
     let isDaemon = results.Contains <@ Arguments.Daemon @>
 
+    let partitionResults (results: #seq<ProcessResult>) =
+        (([], [], [], []), results)
+        ||> Seq.fold (fun (oks, ignores, unchanged, errors) next ->
+            match next with
+            | ProcessResult.Formatted x -> (x :: oks, ignores,unchanged, errors)
+            | ProcessResult.Ignored i -> (oks, i :: ignores, unchanged, errors)
+            | ProcessResult.Unchanged u -> (oks, ignores, u :: unchanged, errors)
+            | ProcessResult.Error(file, e) -> (oks, ignores, unchanged, (file, e) :: errors))
+
+    let reportOnResults (results: #seq<ProcessResult>) =
+        let oks, ignored, unchanged, errored = partitionResults results
+
+        let summaryMessage =
+            $"Formatted: %d{oks.Length}, Ignored : %d{ignored.Length}, Unchanged : %d{unchanged.Length}, Errored: %d{errored.Length}"
+
+        stdlog summaryMessage
+
+        errored
+        |> Seq.iter (fun (file, error) -> elog $"Failed to format file: {file} : {error}")
```

Comment:
Should match on the exception. If it's a fantomas exception then should the error message without a stacktracke. Otherwise if it's verbose then put the the whole exception in, even if it's not from fantomas.